glpi-project/glpi
# CSS Compile Improvements

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

- Change palettes to use CSS variables rather than SCSS variables.
- Remove all imports of tabler/bootstrap/main GLPI CSS from palettes
- Change GLPI CSS customizations to use the CSS variables rather than the SCSS variables (ex: --tblr-primary instead of $primary).
- Upgrade Tabler - Required to do this now as more recent updates finally made progress towards using the CSS variables whenever possible rather than SCSS variables.

Extra:
- Core palettes are now combined into a single CSS file and can be dynamically switched simply by changing the existing attributes on the root HTML element. Changing the palette selection in the user preferences can now let the user see a preview of the theme by changing these attributes. This more importantly reduces the amount of compiling needed for the palettes.
- To make UI work easier, a new debug widget was added to be able to change the palette/theme for the current page.

These changes will greatly reduce the amount of time it takes to compile the CSS.
Execution time of `bin/console build:compile_scss`:
- `main` branch: 6 minutes
- This PR so far: 30 seconds (12x faster for me)
